registry/auth.go

```diff
@@ -74,6 +74,12 @@ func (auth *RequestAuthorization) getToken() (string, error) {
 	return "", nil
 }
 
+// Checks that requests to the v2 registry can be authorized.
+func (auth *RequestAuthorization) IsAuthorized() bool {
```

Comment:
> We could get some additional coverage by making "IsAuthorized" require a "200 OK" from "/v2/". This would partially cover the case of sending "Basic" or "Bearer" to the registry.

I agree but should only be needed for basic auth. I don't think we need to cover the case here where the token returned from the auth server does not contain the required scopes.